src/pages/patientView/PatientViewGenePanelModal/GenesList.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import { GenePanel, GenePanelToGene } from 'shared/api/generated/CBioPortalAPI';
+import { observer } from 'mobx-react';
+import { SimpleCopyDownloadControls } from 'shared/components/copyDownloadControls/SimpleCopyDownloadControls';
+import { serializeData } from 'shared/lib/Serializer';
+import './styles.scss';
+import { chunk } from 'lodash';
+
+interface IGenesListProps {
+    genePanelCache: GenePanel;
+    panelName: string;
+    columns?: number;
+    id?: string | undefined;
+}
+
+interface IGenesListState {
+    filter: string;
+}
+
+@observer
+export default class PatientViewGenePanelModal extends React.Component<
+    IGenesListProps,
+    IGenesListState
+> {
+    constructor(props: IGenesListProps) {
+        super(props);
+        this.state = {
+            filter: '',
+        };
+    }
+
+    handleChangeInput = (value: string) => {
+        this.setState({ filter: value });
+    };
+
+    filterGenes = () => {
+        const { genes } = this.props.genePanelCache;
+        const { filter } = this.state;
+        if (filter) {
+            const lowerCaseFilter = filter.toLowerCase();
+            return genes.filter(
+                each =>
+                    each.entrezGeneId.toString().includes(lowerCaseFilter) ||
+                    each.hugoGeneSymbol.toLowerCase().includes(lowerCaseFilter)
+            );
+        }
+        return genes;
+    };
+
+    genesDividedToColumns = (genes: GenePanelToGene[]) => {
+        return chunk(genes, this.genesCountPerColumn(genes.length));
+    };
+
+    renderColumns = () => {
+        const filtered = this.filterGenes();
+        if (filtered.length === 0) {
+            return <td>No matches.</td>;
+        }
+        return this.genesDividedToColumns(filtered).map(col => (
+            <td>
+                {col.map(each => (
+                    <p key={each.entrezGeneId}>{each.hugoGeneSymbol}</p>
+                ))}
+            </td>
+        ));
+    };
+
+    getDownloadData = () => {
+        const downloadData = [['Genes'], ...this.genesDividedToRows()];
+        return serializeData(downloadData);
+    };
+
+    genesDividedToRows = () => {
+        const { genes } = this.props.genePanelCache;
+        const genesByColumns = this.genesDividedToColumns(genes);
+        const genesByRows = [];
+        for (let i = 0; i < this.genesCountPerColumn(genes.length); i++) {
+            const genesPerRow = [];
+            for (let j = 0; j < this.columnsCount(); j++) {
+                genesPerRow.push(
+                    genesByColumns[j] && genesByColumns[j][i]
+                        ? genesByColumns[j][i].hugoGeneSymbol
+                        : ''
+                );
+            }
+            genesByRows.push(genesPerRow);
+        }
+        return genesByRows;
+    };
+
+    columnsCount = () => {
+        return this.props.columns || 1;
+    };
+
+    genesCountPerColumn = (totalLength: number) => {
```

Comment:
geneCountPerColumn        gene should not have s following.  minor but better english.